src/python/pants/engine/target.py

```diff
@@ -1592,6 +1596,69 @@ class ExplicitlyProvidedDependencies:
     includes: FrozenOrderedSet[Address]
     ignores: FrozenOrderedSet[Address]
 
+    @memoized_method
+    def any_are_covered_by_includes(self, addresses: tuple[Address, ...]) -> bool:
+        """Return True if every address is in the explicitly provided includes.
+
+        Note that if the input addresses are file addresses, they will still be marked as covered if
+        their original BUILD target is in the explicitly provided includes.
+        """
+        return any(
+            addr in self.includes or addr.maybe_convert_to_build_target() in self.includes
+            for addr in addresses
+        )
+
+    @memoized_method
+    def remaining_after_ignores(self, addresses: tuple[Address, ...]) -> set[Address]:
+        """All addresses that are not covered by the explicitly provided ignores.
+
+        Note that if the input addresses are file addresses, they will still be marked as covered if
+        their original BUILD target is in the explicitly provided ignores.
+        """
+        return {
+            addr
+            for addr in addresses
+            if (
+                addr not in self.ignores
+                and addr.maybe_convert_to_build_target() not in self.ignores
+            )
+        }
+
+    def maybe_warn_of_ambiguous_dependency_inference(
+        self,
+        ambiguous_addresses: tuple[Address, ...],
+        original_address: Address,
+        *,
+        context: str,
+        import_reference: str,
+    ) -> None:
+        """If the module is ambiguous and the user did not disambiguate via explicitly provided
+        dependencies, warn that dependency inference will not be used."""
+        if not ambiguous_addresses or self.any_are_covered_by_includes(ambiguous_addresses):
+            return
+        remaining_after_ignores = self.remaining_after_ignores(ambiguous_addresses)
+        if len(remaining_after_ignores) <= 1:
+            return
+        logger.warning(
+            f"{context}, but Pants cannot safely infer a dependency because >1 target owns "
```

Comment:
`>1` -> `more than one` -- more accessible language usage

Reply:
I love it, thanks!